src/datasets/arrow_dataset.py

```diff
@@ -3949,60 +3949,63 @@ def shards_with_embedded_external_files(shards):
             shards = shards_with_embedded_external_files(shards)
 
         files = api.list_repo_files(repo_id, repo_type="dataset", revision=branch, token=token)
-        files = [file for file in files if file.startswith("data/")]
+        data_files = [file for file in files if file.startswith("data/")]
 
-        def path_in_repo(_index):
-            return f"data/{split}-{_index:05d}-of-{num_shards:05d}.parquet"
+        def path_in_repo(_index, shard):
+            return f"data/{split}-{_index:05d}-of-{num_shards:05d}-{shard._fingerprint}.parquet"
```

Comment:
Maybe you can check if the first shard is already on the Hub, and show a warning that says that it's resuming the push_to_hub operation ?